cocotb/clock.py

```diff
@@ -165,3 +168,134 @@ async def start(self, cycles=None, start_high=True):
 
     def __str__(self):
         return type(self).__qualname__ + "(%3.1f MHz)" % self.frequency
+
+
+class SimClock(BaseClock):
+    r"""High performance, configurable simulator clock driver.
+
+    This class is a wrapper for a C++ layer clock driver with methods for control.
+    Use :meth:`start` to start the clock driver and :meth:`stop` to stop it.
+    The :attr:`is_running` property returns whether the driver is active.
+
+    Example:
+
+    .. code-block:: python
+
+        # defaults to 50:50 duty cycle
+        c = SimClock(dut.clk, 100, units='ns')
+        c.start()
+
+    Args:
+        signal: The clock pin/signal to be driven.
+        period (int, float, 2-tuple): The clock period. Must convert to an even number of
+            timesteps. Provide a 2-tuple as (high_time, low_time) to set the duty cycle manually.
+        units (str, optional): One of
+            ``'step'``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
+            When *units* is ``'step'`` (the default),
+            the timestep is determined by the simulator (see :make:var:`COCOTB_HDL_TIMEPRECISION`).
+        duty_cycle (float, optional): The clock duty cycle as a float in [0, 1]. Default is 0.5.
+            Not compatible with 2-tuple clock period, with the 2-tuple taking precedence.
+        jitter (int, float, optional): Maximum jitter for one clock period, in same units as period.
+            Default is 0.
+
+    If clock jitter is non-zero, random edge-to-edge clock jitter will be generated
+    using a normal (gaussian) distribution for the jitter values.
+    With a clock period defined as one rising edge to the next rising edge, the jitter
+    will be bounded each clock period by ``[-jitter, +jitter]``. Jitter example:
+
+    .. code-block:: python
+
+        # 1MHz clock with 10ns of jitter
+        c = SimClock(dut.clk, 1000, units='ns', jitter=10)
+        c.start()
+
+    Raises:
+        ValueError: If tuple is used for clock period and not a 2-tuple,
+            or duty cycle value is not in [0, 1].
+
+    .. versionadded:: 1.5
+    """
+    def __init__(
+        self, signal: SimHandleBase,
+        period: Union[Union[int, float], Tuple[Union[int, float], Union[int, float]]],
+        units: str = 'step', duty_cycle: Optional[float] = None,
+        jitter: Union[int, float] = 0
+    ):
+        super().__init__(signal)
+        if isinstance(period, tuple):
+            if len(period) != 2:
+                raise ValueError("Clock period tuple must have 2 members")
+            if duty_cycle is not None:
+                warnings.warn("Duty cycle cannot be used with clock period tuple. "
+                              "Duty cycle value will be ignored", category=RuntimeWarning)
+
+            high_steps = get_sim_steps(period[0], units)
```

Comment:
```suggestion
            # this naming applies to the default `start_high=True`
            high_steps = get_sim_steps(period[0], units)
```
I guess internally, we can normalize the wording like here, but toward the user, we may want to speak of "first part of the period" etc.?

Reply:
See my comment above. High is always the first part of the defined period, `start_high` just decides if we start with the high part or start with the low part.